### DISCUSSION TITLE
TODO for Revamp Handler branch

### DISCUSSION THREAD

**Discussion by jacoobes** [sern-handler/handler#45] [Announcements]
# Need to push the revamp handler out soon.

##  And here are some things todo;

- [x] Event emitters for commands, errors, and anything else. 

- [x] Change to default exports

- [x] Finish event plugins handling

- [x] Fix linting issue, will be fixed when rebase occurs

- [x] Move many important pieces of code to index.ts for exporting

- [ ] Testing in JavaScript and TypeScript

- [x] Modal and Autocomplete interaction support

- [x] Refactoring Context.ts

- [x] Making command properties optional to reduce bloat

**Comment by jacoobes** [comment]
Context.ts refactoring will be moved to another sem minor!

**Comment by jacoobes** [comment]
Linting issue will be fixed when we merge!

**Reply by Murtatrxx** [reply]
We merged.

**Reply by Murtatrxx** [reply]
Now what?

**Reply by jacoobes** [reply]
visit api-update branch, that's what's next

**Comment by Murtatrxx** [comment]
Could I close this?
